test/runtests.jl

```diff
@@ -407,6 +408,27 @@ function test_potential_energy_equation_terms(model; geopotential_height = nothi
 
     return nothing
 end
+function test_PEbuoyancytracer_PElineareos(grid)
+
+    model_buoyancytracer = NonhydrostaticModel(; grid, buoyancy=BuoyancyTracer(), tracers=:b)
+    model_lineareos = NonhydrostaticModel(; grid, buoyancy=SeawaterBuoyancy(), tracers=(:S, :T))
+    C_grad(x, y, z) = 0.01 * z
+    set!(model_lineareos, S = C_grad, T = C_grad)
+    linear_eos_buoyancy(grid, buoyancy, tracers) =
+        KernelFunctionOperation{Center, Center, Center}(buoyancy_perturbationᶜᶜᶜ, grid, buoyancy, tracers)
+    b_field = Field(linear_eos_buoyancy(model_lineareos.grid, model_lineareos.buoyancy.model, model_lineareos.tracers))
+    compute!(b_field)
+    set!(model_buoyancytracer, b = interior(b_field))
+    pe_buoyancytracer = Field(PotentialEnergy(model_buoyancytracer))
+    compute!(pe_buoyancytracer)
+    pe_lineareos = Field(PotentialEnergy(model_lineareos))
+    compute!(pe_lineareos)
+
+    @test all(interior(pe_buoyancytracer) .== interior(pe_lineareos))
```

Comment:
I guess we're only testing CPU architecture in this function, but I think this will fail on GPUs. I'm okay leaving it like this since I think `PotentialEnergy()` is tested in a way that's compatible with GPUs somewhere else. It is, right?

Reply:
Relatedly, I'm a little out of the loop about buoyancy models. This is all the models we currently have available, correct?

https://github.com/tomchor/Oceanostics.jl/blob/d2060579e175cd4ac3265a165cfd49d2ac304355/test/runtests.jl#L526-L528

Reply:
Yes I think this all buoyancy models but it does not test all the [second order `RoquetEquationOfState`s](https://clima.github.io/SeawaterPolynomials.jl/dev/API/#SeawaterPolynomials.SecondOrderSeawaterPolynomials.RoquetSeawaterPolynomial) which we can do but I thought two would be enough.

Reply:
> I guess we're only testing CPU architecture in this function, but I think this will fail on GPUs. I'm okay leaving it like this since I think `PotentialEnergy()` is tested in a way that's compatible with GPUs somewhere else. It is, right?

Yes the function setup and computation tested is tested here 
https://github.com/tomchor/Oceanostics.jl/blob/d2060579e175cd4ac3265a165cfd49d2ac304355/test/runtests.jl#L387-L390
but it only checks that it can set the `Field` and can then `compute!` the `Field` i.e. not testing it against another value.
I will try and test on a `GPU()` today.

Reply:
> I guess we're only testing CPU architecture in this function, but I think this will fail on GPUs. I'm okay leaving it like this since I think `PotentialEnergy()` is tested in a way that's compatible with GPUs somewhere else. It is, right?

This test passed when I ran it on the GPU, so I think we should be ok!